src/alerts/check_signals_updates.R

```diff
@@ -0,0 +1,207 @@
+}
+
+#' Takes the response from a GitHub Actions run of a single indicator
+#' and outputs a status message to be posted to Slack
+#'
+#' @param response JSON payload returned from `gh_status`
+#' @param indicator_id ID of the indicator
+#'
+#' @returns String status message to be posted to Slack
+slack_build_workflow_status <- function(response, indicator_id) {
+  base_logs_url <- paste0("https://github.com/ocha-dap/hdx-signals/actions/runs/")
+
+  if (status_code(response) == 200) {
+    content <- content(response, as = "text", encoding = "UTF-8")
+    json_data <- fromJSON(content, flatten = TRUE)
+    df_runs <- as.data.frame(json_data)
+    df_runs$date <- as.Date(df_runs$workflow_runs.created_at)
+
+    # Get today's scheduled runs from the main branch
+    df_sel <- subset(
+      df_runs,
+      workflow_runs.event == "schedule" &
+        workflow_runs.head_branch == "main" &
+        date == Sys.Date()
+    )
+
+    if (nrow(df_sel) == 1) {
+      status <- df_sel[1, ]$workflow_runs.conclusion
+      run_id <- df_sel[1, ]$workflow_runs.id
+      if (status == "failure") {
+        run_link <- paste0(base_logs_url, run_id)
+        status_update <- paste0(":red_circle: ", indicator_id, ": Failed update - <", run_link, "|Check logs> \n")
+      } else if (status == "success") {
+        status_update <- paste0(":large_green_circle: ", indicator_id, ": Successful update \n")
+      }
+      # If no scheduled runs happened off of main today
+    } else if (nrow(df_sel) == 0) {
+      status_update <- paste0(":heavy_minus_sign: ", indicator_id, ": No scheduled update \n")
+    } else {
+      status_update <- paste0(":red_circle: ", indicator_id, ": More than one scheduled run today \n")
+    }
+  } else {
+    status_update <- paste0(
+      ":red_circle: ",
+      ind,
+      ": Failed request for workflow status - ",
+      status_code(response),
+      "\n"
+    )
+  }
+
+  return(status_update)
+}
+
+indicators_path <- "src/indicators"
+indicators <- list.files(indicators_path)[file.info(file.path(indicators_path, list.files(indicators_path)))$isdir]
+full_status <- ""
+n_signals <- 0
+signals <- ""
+
+# TODO: Hard coding this because the monitoring scripts don't totally match the indictor ids
+indicators_azure <- c(
+  "acled_conflict",
+  "idmc_displacement_conflict",
+  "idmc_displacement_disaster",
+  "ipc_food_insecurity",
+  "jrc_agricultural_hotspots"
+)
+
+log_info("Checking GitHub Actions status...")
+for (ind in indicators) {
+  gh_response <- gh_status(ind)
+  workflow_status <- slack_build_workflow_status(gh_response, ind)
+  full_status <- paste0(full_status, workflow_status)
+}
+log_info("Successfully checked")
+
+log_info("Checking for signals across all indicators...")
+for (ind in indicators_azure) {
+  test <- Sys.getenv("TEST", unset = TRUE)
+  fn_signals <- paste0(
+    "output/",
+    ind,
+    if (test) "/test" else "",
+    "/signals.parquet"
+  )
+  df <- cs$read_az_file(fn_signals)
+  if (nrow(df) > 0) {
```

Comment:
In our runs, all rows of the campaign data frame will go into the same campaign, so a single URL. So maybe just need to get the # of campaigns and iso3 codes for the campaigns to report in the Slack message?